62.CAD/common.hlsl

```diff
@@ -22,10 +24,53 @@ struct DrawObject
     uint64_t geometryAddress;
 };
 
+template <typename float_t>
+struct QuadBezierAnalyticArcLengthCalculator
+{
+#ifdef __cplusplus
+    using float2_t = nbl::core::vector2d<float_t>;
+#else
+    using float2_t = vector<float_t, 2>;
+#endif
+
+    static QuadBezierAnalyticArcLengthCalculator construct(float_t lenA2, float_t AdotB, float_t a, float_t b, float_t  c, float_t b_over_4a)
+    {
+        QuadBezierAnalyticArcLengthCalculator ret = { lenA2, AdotB, a, b, c, b_over_4a };
+        return ret;
+    }
+    
+    static QuadBezierAnalyticArcLengthCalculator construct(float2_t P0, float2_t P1, float2_t P2)
+    {
+        QuadBezierAnalyticArcLengthCalculator ret;
+        float2_t A = P0 - 2.0 * P1 + P2;
+        float2_t B = 2.0 * (P1 - P0);
+        float2_t C = P0;
```

Comment:
now that this struct is in `Quadratic` struct and named `AnalyticArcLenCalculator`
get these values directly by getting `Quadratic<>` instead of P0,P1,P2 here

Reply:
done